### PR TITLE
[ 3 ptx] My reservations#6

### LINKED ISSUE #6: [ 3 ptx] My reservations
Allow a user to reserve a car

### PR DESCRIPTION
**Pull request by lojacho**
In this pull request, I was able to: 

- Fetch reservations for a specific user: 
getbyuser

- Fetch reservations  for reservation id
getbyid

Create a reservation : 
Post

Thanks for the review.
Diego

### CODE REVIEW

File: app/controllers/api/v1/reservations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+class Api::V1::ReservationsController < ApplicationController
+  before_action :find_reservation, only: %i[show update destroy]
+  # GET /reservations by user
+  def index
+    @reservations = Reservation.all
```
Comment (paulsaenzsucre) [inline app/controllers/api/v1/reservations_controller.rb]: Hi, @dgcuenca:
Great work so far!!!

I think that you only need to return the reservations of the selected user.

The rationale behind this is that in our routes we have `reservations` nested inside `users`, so you have for the index action the following route:

`/api/v1/users/:user_id/reservations(.:format)`

So, please kindly update this or let me know if you have any concerns about it.

**Review by paulsaenzsucre — CHANGES_REQUESTED** [review]
STATUS: CHANGES REQUIRED
Hi @dgcuenca,



Good job so far!
To Highlight! 👏 🟢
✔️ Required gem used and needed dependencies added
✔️ Your code looks clean and organized
✔️ Good use of parent's classes method
✔️ API endpoint created
✔️ Required controller and model added
✔️ Apps work as required

There are some issues that you still need to work on to go to the next project but you are almost there!

Required Changes ♻️
Check the comments under the review.

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

You can also consider:

N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: app/controllers/api/v1/reservations_controller.rb (hunk 1 above)
Comment (lojacho) [reply app/controllers/api/v1/reservations_controller.rb]: Hello @paulsaenzsucre 

Changes where implemented, thanks for the feedback.

File: app/controllers/api/v1/reservations_controller.rb [hunk 2]
```diff
@@ -2,7 +2,8 @@ class Api::V1::ReservationsController < ApplicationController
   before_action :find_reservation, only: %i[show update destroy]
   # GET /reservations by user
   def index
-    @reservations = Reservation.all
+    @current_user = User.find(request.params['user_id'])
```
Comment (paulsaenzsucre) [inline app/controllers/api/v1/reservations_controller.rb:5]: Great job so far!!!

This is OPTIONAL: `current_user` is already a name from Devise, so you can consider use another like @user, or other.

**Review by paulsaenzsucre — APPROVED** [review]
Hi @dgcuenca, 
Good job so far! 

May I highlight the following:

- You implemented the required changes

## STATUS: APPROVED :muscle::clap:

image

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous review
 
 ## Optional suggestions
 
 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @paulsaenzsucre  if something is not 100% clear.

**Pull request merged**